subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -408,4 +413,78 @@ public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setExportKeys(true);
         }
     }
+
+    public static class ConfigurationCacheOption extends BooleanBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache";
+        public static final String LONG_OPTION = "configuration-cache";
+
+        public ConfigurationCacheOption() {
+            super(PROPERTY_NAME, BooleanCommandLineOptionConfiguration.create(
+                LONG_OPTION,
+                "Enables the configuration cache. Gradle will try to reuse the build configuration from previous builds.",
+                "Disables the configuration cache."
+            ).incubating());
+        }
+
+        @Override
+        public void applyTo(boolean value, StartParameterInternal settings, Origin origin) {
+            settings.setConfigurationCacheEnabled(value);
+        }
+    }
+
+    public static class ConfigurationCacheFailOnProblemsOption extends BooleanBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache.fail-on-problems";
+        public static final String LONG_OPTION = "configuration-cache-fail-on-problems";
+
+        public ConfigurationCacheFailOnProblemsOption() {
+            super(PROPERTY_NAME, BooleanCommandLineOptionConfiguration.create(
+                LONG_OPTION,
+                "Lets the configuration cache fail on problems. This is the default.",
+                "Lets the configuration cache do not fail on problems."
+            ).incubating());
+        }
+
+        @Override
+        public void applyTo(boolean value, StartParameterInternal settings, Origin origin) {
+            settings.setConfigurationCacheFailOnProblems(value);
+        }
+    }
+
+    public static class ConfigurationCacheMaxProblemsOption extends IntegerBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache.max-problems";
+        public static final String LONG_OPTION = "configuration-cache-max-problems";
```

Comment:
This could be changed to `--max-configuration-cache-problems`

Reply:
I wouldn't add a command line option for this until we have some feedback that we need it. Probably the same for options that control what to do when there are problems (ignore, fail, whatever).

Reply:
Command line option removed, gradle and system property kept